src/python/dxpy/bindings/dxfile.py

```diff
@@ -363,6 +396,16 @@ def write(self, data, multithread=True, **kwargs):
             does not affect where the next :meth:`write` will occur.
 
         '''
+        self.file_limits = dxpy.api.project_describe(self.project, {'fields': {'fileUploadParameters': True}})['fileUploadParameters']
+        self._set_file_limits(self.file_limits)
+
+        self._write_buf_size = self._get_buffer_size_for_file(self.file_size, self.file_is_mmapd)
```

Comment:
Yes, you're correct (on both counts).

I'm fine with changing the existing functionality since this should only convert previously broken code (code that specified a bogus buffer size) to now working code... right?